### DISCUSSION TITLE
Migrating from CircleCI

### DISCUSSION THREAD

**Discussion by malimccalla** [community/community#26975] [Actions]
<p>Hey everyone,</p>
<p>I currently have a CircleCi config file that builds my code and runs my tests for the client and the api. I would really like to move over to Github actions to do all this but I’ve been having issues migrating over.</p>
<p>The services I currently use are</p>
<ul>
<li>
<p>Postgres</p>
</li>
<li>
<p>Redis</p>
</li>
<li>
<p>Cypress </p>
</li>
</ul>
<p>I’ve looked at a bunch of exaples and read the docs but it seems more complex than I was hoping for. E.g The <a href="https://github.com/actions/example-services/blob/master/.github/workflows/postgres-service.yml" rel="nofollow noopener">postgres example service </a> alone is over 70 lines of code when my original CI config is just 40.</p>
<p>Here is my CircleCi config, I would love to know how to achieve the same with Github Actions</p>
<pre><code>version: 2
jobs:
  build:
    docker:
      - image: cypress/base:10
      - image: circleci/postgres:9.6.5-alpine-ram
        environment:
          POSTGRES_USER: root
          POSTGRES_DB: tunebase-test
      - image: redis

    working_directory: ~/tunebase

    steps:
      - checkout

      # Download and cache dependencies
      - restore_cache:
          name: Restore Yarn Package Cache
          keys:
            - v24-dependencies-{{ checksum "yarn.lock" }}
            # fallback to using the latest cache if no exact match is found
            - v24-dependencies-

      - run: yarn install --frozen-lockfile
      - run: yarn run bootstrap

      - save_cache:
          name: Save Yarn Package Cache
          paths:
            - ~/.cache
          key: v24-dependencies-{{ checksum "yarn.lock" }}

      # Make sure both api and www build
      - run: yarn run build

      # run client and api tests
      - run: yarn --cwd packages/api test
      - run: yarn --cwd packages/www test:ci
</code></pre>
<p>Thanks!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/malimccalla">@malimccalla</a> ,</p>
<p>It should be like below:</p>
<pre><code>name: version2
on: [push] # define the event to trigger the workflow

jobs:
  job1:
    defaults:
      run:
        working-directory: ~/tunebase #set the working directory for the job.
    services: #define 3 different services
      cypress:
        image: cypress/base:10
      postgres:
        image: circleci/postgres:9.6.5-alpine-ram
        env:
          POSTGRES_USER: root
          POSTGRES_PASSWORD: password
          POSTGRES_DB: tunebase-test
      redis:
        image: redis

    runs-on: ubuntu-latest
    steps:
      - name: checkout # use action to checkout the repo content
        uses: actions/checkout@v2
      - name: Restore Yarn Package Cache # restore &amp; save cache
        uses: actions/cache@v1
        with:
          path: ~/.cache
          key: v24-dependencies-{{ checksum "yarn.lock" }}
          restore-keys: |
            v24-dependencies-{{ checksum "yarn.lock" }}
            v24-dependencies-
       - run: yarn install --frozen-lockfile
       - run: yarn run bootstrap

       # Make sure both api and www build
       - run: yarn run build
</code></pre>
<p>Please read below docs for more details:</p>
<p>1. <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices</a></p>
<p>2. <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows</a></p>
<ol start="3">
<li>Currently yaml archor is not supported in Github actions.</li>
</ol>
<p>Thanks.</p>
